### ISSUE TITLE
Ween trial's --dry-run away from the deprecated TestCase.visitor code.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4063]
|<img alt="kelly's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| kelly reported|
|-|-|
|Trac ID|trac#4063|
|Type|defect|
|Created|2009-10-24 13:38:22Z|
|Branch|https://github.com/twisted/twisted/tree/dry-run-4063|

t.t.runner.TrialRunner is still using the deprecated visitor code to do `--dry-run`.

Attachments:

* [fix-dry-run.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/707/7072c31ed58b3a21bbf880c206802aa5221eecfb/6bda2f17744c8660bdf6b1cd18f3bc75a131fa4f.patch) (2534 bytes) - added by kelly on 2009-10-24 13:39:22Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4063 4063
type__defect defect
reporter__kelly kelly
priority__normal normal
milestone__ 
branch__branches_dry_run_4063 branches/dry-run-4063
branch_author__kelly__thijs kelly, thijs
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1256391502000000 1256391502000000
changetime__1258329961000000 1258329961000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [27434]) Branching to 'dry-run-4063'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

Thanks kelly, I branched in r27435

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [27437]) Add copyright header, refs [#4063](https://github.com/twisted/twisted/issues/4063)

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml set owner to @thijstriemstra|
|-|-|

There's already a method that does this called `_iterateTests`, which lives in `twisted.trial.util`. Can you please simply change dry-run to use that?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [27506]) Remove _depthFirst and use _iterateTests

Refs [#4063](https://github.com/twisted/twisted/issues/4063)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @jml<br>@exarkun set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_8"></a>|@jml set owner to @exarkun|
|-|-|

Thanks Jean-Paul! Please land this with my blessing.

Thanks to kelly & thijs too.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [27510]) Merge dry-run-4063

Author: kelly, thijs, exarkun
Reviewer: jml
Fixes: [#4063](https://github.com/twisted/twisted/issues/4063)

Stop using the deprecated `DryRunVisitor` to implement the
`trial --dry-run` feature.  Instead, use a private trial
helper to iterate over all of the test cases and start/succeed/stop
them.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
